packages/compiler-cli/test/ngtsc/ngtsc_spec.ts

```diff
@@ -3243,6 +3243,214 @@ runInEachFileSystem(os => {
       });
     });
 
+    describe('aliasing re-exports', () => {
+      it('should re-export a directive from a different file under a private symbol name', () => {
+        env.write('dir.ts', `
+          import {Directive} from '@angular/core';
+
+          @Directive({
+            selector: 'dir',
+          })
+          export class Dir {}
+        `);
+        env.write('module.ts', `
+          import {NgModule} from '@angular/core';
+          import {Dir} from './dir';
+
+          @NgModule({
+            declarations: [Dir],
+            exports: [Dir],
+          })
+          export class Module {}
+        `);
+
+        env.driveMain();
+        const jsContents = env.getContents('module.js');
+        const dtsContents = env.getContents('module.d.ts');
+
+        expect(jsContents).toContain('export { Dir as ɵngExportɵModuleɵDir } from "./dir";');
+        expect(dtsContents).toContain('export { Dir as ɵngExportɵModuleɵDir } from "./dir";');
+      });
+
+      it('should re-export a directive from an exported NgModule under a private symbol name',
+         () => {
+           env.write('dir.ts', `
+          import {Directive, NgModule} from '@angular/core';
+
+          @Directive({
+            selector: 'dir',
+          })
+          export class Dir {}
+
+          @NgModule({
+            declarations: [Dir],
+            exports: [Dir],
+          })
+          export class DirModule {}
+        `);
+           env.write('module.ts', `
+          import {NgModule} from '@angular/core';
+          import {DirModule} from './dir';
+
+          @NgModule({
+            exports: [DirModule],
+          })
+          export class Module {}
+        `);
+
+           env.driveMain();
+           const jsContents = env.getContents('module.js');
+           const dtsContents = env.getContents('module.d.ts');
+
+           expect(jsContents).toContain('export { Dir as ɵngExportɵModuleɵDir } from "./dir";');
+           expect(dtsContents).toContain('export { Dir as ɵngExportɵModuleɵDir } from "./dir";');
+         });
+
+      it('should not re-export a directive that\'s already exported', () => {
+        env.write('dir.ts', `
+          import {Directive} from '@angular/core';
+
+          @Directive({
+            selector: 'dir',
+          })
+          export class Dir {}
+        `);
+        env.write('module.ts', `
+          import {NgModule} from '@angular/core';
+          import {Dir} from './dir';
+
+          @NgModule({
+            declarations: [Dir],
+            exports: [Dir],
+          })
+          export class Module {}
+
+          export {Dir};
+        `);
+
+        env.driveMain();
+        const jsContents = env.getContents('module.js');
+        const dtsContents = env.getContents('module.d.ts');
+
+        expect(jsContents).not.toContain('export { Dir as ɵngExportɵModuleɵDir } from "./dir";');
```

Comment:
I would make these assertions less susceptible to potential formatting changes:
```suggestion
        expect(jsContents).not.toContain('ɵngExportɵModuleɵDir');
```